policy-controller/grpc/src/outbound.rs

```diff
@@ -294,13 +338,13 @@ fn to_service(
     }
 }
 
-fn timestamp_then_name<LM, LR, RM, RR>(
-    (left_id, left_route): &(GroupKindNamespaceName, OutboundRoute<LM, LR>),
-    (right_id, right_route): &(GroupKindNamespaceName, OutboundRoute<RM, RR>),
+fn timestamp_then_name<L: Route, R: Route>(
```

Comment:
I think now we never compare routes of different types, so we can simplify this to only use one type parameter.

Reply:
This will enforce, at the type level, that routes cannot be mixed.